api/nodes/views.py

```diff
@@ -1952,6 +1954,23 @@ def get_default_queryset(self):
     def get_queryset(self):
         return self.get_queryset_from_request().distinct()
 
+    def get_queryset_from_request(self):
```

Comment:
I don't think `get_queryset_from_request` needs to be overridden here. All of this logic could be placed in `get_queryset`.